samcli/lib/sync/flows/layer_sync_flow.py

```diff
@@ -341,3 +350,17 @@ def gather_dependencies(self) -> List["SyncFlow"]:
 
     def _equality_keys(self) -> Any:
         return self._function_identifier, self._layer_arn, self._new_layer_version
+
+    def _verify_function_status(self) -> bool:
```

Comment:
I think these functions are the same (the one in function sync flow), can we move them in a common place?

Reply:
This is a non-static method that needs to be class specific. However the only common inheritance between those two classes would be SyncFlow. I don't think we should put in function specific method into the general SyncFlow class. However I can propose a new common class that both FunctionSyncFlow and FunctionLayerReferenceSync inherit. Do you think that would be a better solution?

Reply:
Or another option, we can make this method static and add parameters for lambda client and function identifier (the only fields that is required from `self`).

I do agree for not putting this into generic sync flow class.

Reply:
Yep static method with function identifier and lambda client passed in also works! I think this is minimum change here. I'll go with this route.